### PR TITLE
Fix/docker

### PR DESCRIPTION
**Pull request by Brandhoej**
## Description
Docker was not compatible for Fastify default host 'localhost' and instead it required '0.0.0.0'. This has been fixed in this PR by allowing an environment variable to define this value. Additonally, we made the .env file optional as docker-compose should not be required to define such a file. This was fixed by removing the process exit if the parsing failed.

### Type of change
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Feature that assists further development (non-breaking change for the developer)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?
This was tested with a docker-compose configuration and lcoally with npm run start.

### Checklist:

- [X] You have added a clear title and description for the PR
- [X] The CI pass
- [ X There are no merge conflicts
- [X] The code satisfies the [code standards](https://docs.google.com/document/d/1R9n9MeuRwz3WSrnWZUjrbmGeosTlu1Xz8f5F7gu2AJE/edit#heading=h.e9xl4xfoldow) 
- [X] I have added docstrings for my code, where needed
- [X] Code implements intended functionality
- [X] All relevant tests pass (Black, White, Integration, Monkey, etc.)

### CODE REVIEW

**Review by Dieuer — APPROVED** [review]
Good job Andreas :)

File: src/presentation/exercise_controller.ts [hunk 1]
```diff
@@ -16,13 +16,13 @@ export async function exerciseController(
...
+          })
           .catch((error) => {
             // Promise rejected or an exception was thrown
             reply.status(500).send(error);
-          })
-          .then((resolve: void | IGetAllExercisesResponse) => {
-            // Promise resolved with a response
-            reply.status(200).send(resolve);
           });
       } catch (error) {
```
Comment (ZeniaHansen) [inline src/presentation/exercise_controller.ts:15]: Hvorfor blande async/await med promise .then .catch? Bare brug try/catch og await de ting der står i .then.

File: src/presentation/exercise_controller.ts [hunk 2]
```diff
@@ -45,20 +45,20 @@ export async function exerciseController(
...
               // The exercise was found so we return 200 OK
-              reply.status(200).send(response);
+              reply.status(200).send(response.exercise);
             }
+          })
+          .catch((error) => {
+            // Promise was rejected or an exception was thrown
+            reply.status(500).send(error);
           });
       } catch (error) {
```
Comment (ZeniaHansen) [inline src/presentation/exercise_controller.ts:42]: Samme her. Bør nok være async/await det hele.

**Review by ZeniaHansen — APPROVED** [review]
Et par kommentarer, men ikke så vigtigt. Tænker det er fint.

**Pull request merged**